Fix bug in non_blocking copy

li-roy opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#20305 Fix bug in non_blocking copy**

Differential Revision: [D15276094](https://our.internmc.facebook.com/intern/diff/D15276094)

Commit:
Fix bug in non_blocking copy

colesbury commented:
I think you can test this with something like the following pseudo-code:

```
x = torch.empty(10, pinned_memory=True)
y = torch.randn(10, device='cuda')
torch.cuda._sleep(...)
event = torch.cuda.Event().record()
x.copy_(y, non_blocking=True)
# check that event hasn't occurred (i.e that the copy is really non-blocking)
# synchronize
# check that x and y are equal
```

ezyang commented:
Yes, a test would be great

Commit:
Update on "Fix bug in non_blocking copy"

Fix bug in non_blocking copy

gh-metadata: pytorch pytorch 20305 gh/li-roy/12/head
